questions/shop.md

```diff
@@ -0,0 +1,165 @@
+To avoid the player from a loop we will remove the tag as we dont need it more.
+
+#### Pre-1.20.5
+
+    # Command blocks
+    tag @p add buyer.netherite
+    execute as @a[tag=buyer.netherite] store result score @s diamonds run clear @s diamond 0
+    give @a[tag=buyer.netherite,scores={diamonds=5..}] netherite_ingot 1
+    clear @a[tag=buyer.netherite,scores={diamonds=5..}] diamond 5
+    tellraw @a[tag=buyer.netherite,scores={diamonds=5..}] {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
+    tellraw @a[tag=buyer.netherite,scores={diamonds=..4}] {"text":"You don't have 5 diamonds","color":"dark_red"}
+    tag @a remove buyer.netherite
+
+Or if you prefer a function:
+
+    # function example:load
+    scoreboard objectives add diamonds dummy
+    
+    # function example:buy/netherite
+    execute store result score @s diamonds run clear @s diamond 0
+    execute if entity @s[scores={diamonds=5..}] run function example:buy/netherite/success
+    execute unless entity @s[scores={diamonds=5..}] run tellraw @s {"text":"You don't have 5 diamonds","color":"dark_red"}
+
+    # function example:buy/netheirte/success
+    give @s netherite_ingot 1
+    clear @s diamond 5
+    tellraw @s {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
+
+> [!NOTE]
+> The function `example:buy/netherite` must be run `as` the player
+
+#### 1.20.5 and above
+
+    # Command blocks
+    tag @p add buyer.netherite
+    execute as @p[tag=buyer.netherite] store result score @s diamonds if items entity @s container.* diamond
+    give @a[tag=buyer.netherite,scores={diamonds=5..}] netherite_ingot 1
+    clear @a[tag=buyer.netherite,scores={diamonds=5..}] diamond 5
+    tellraw @a[tag=buyer.netherite,scores={diamonds=5..}] {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
+    tellraw @a[tag=buyer.netherite,scores={diamonds=..4}] {"text":"You don't have 5 diamonds","color":"dark_red"}
+    tag @a remove buyer.netherite
+    
+Or if you prefer a function:
+
+    # function example:load
+    scoreboard objectives add diamonds dummy
+    
+    # function example:buy/netherite
+    execute store result score @s diamonds if items entity @s container.* diamond
+    execute if entity @s[scores={diamonds=5..}] run function example:buy/netherite/success
+    execute unless entity @s[scores={diamonds=5..}] run tellraw @s {"text":"You don't have 5 diamonds","color":"dark_red"}]
+    
+    # function example:buy/netherite/success
+    give @s netherite_ingot 1
+    clear @s diamond 5
+    tellraw @s {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
+
+> [!NOTE]
+> The function `example:buy/netherite` must be run `as` the player
+
+### Bedrock
+In bedrock we have the `hasitem` argument, so it uses less commands than in Java.
+
+#### With an NPC
+This commands must be run in this order in an NPC
+
+    tag @initiator[hasitem={item=diamond,quantity=5..}] add buy.netherite
```

Comment:
why is this method using a different approach than all others? Wouldn't the same approach as the "non NPC" one below work, just with `@initiator` instead of the tag?

Reply:
Because instead of run as @p and then check the score we can just check the initiator with the item (bc @p[hasitem=…] will select the nearest player with the item independent if its the player that presed the button)

Reply:
I just realized that I will remove the NPC one and add a note saying:

> in order for it to work with npc, change `@p` to `@initiator`

It’s ok like this?

Reply:
that would work for me, yeah.